ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -327,6 +327,30 @@ EOF
   echo "Enabled AWS Spot instances for worker nodes"
 }
 
+# enable_efa_pg_instance_config is an AWS specific option that enables one worker machineset in a placement group and with EFA Network Interface Type, other worker machinesets will be ENA Network Interface Type by default.....
+function enable_efa_pg_instance_config() {
+  local dir=${1}
+
+  if [ ! -f /tmp/yq ]; then
```

Comment:
Is there a specific reason that we only use yq which is located in /tmp ?

Reply:
no, just copied from https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L308

Reply:
Sorry @liangxia Seems yq-go doesn't work, so I change it back.
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/41990/rehearse-41990-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-efa-pg-f14/1689119998305374208
```
environment: line 347: yq-go: command not found
```

Reply:
We should integrate the yq command in the image, rather than maintain it in every function.
curl from github is not reliable, we've been suffering from this kind of issue for several times.

Reply:
Thank you @liangxia Good to know the reason. Integrate the yq command in the image should be a good idea. But how to do that? I didn't find any relevant code in the repository, can you please give some help?

Reply:
This step use the image `installer`. I think it's build from this dockerfile https://github.com/openshift/installer/blob/master/images/installer/Dockerfile.ci

Maybe we can deal with this issue in a separate PR.

Reply:
Thank you @liangxia Seems it's another repository, and I never touched that repository. 
> Maybe we can deal with this issue in a separate PR.

That should be better, and maybe installer QE are more professional for that? @yunjiang29 @jianlinliu

Reply:
Now only `upi-installer` image integrated yq binary, but not `installer` image, we do not have plan to update `installer` image to.

I agree with that, we need to avoid download the binary as much as possible, two alternatives to try:
1. create a dedicated manifest file priory to ipi-install-install tep
2. run `python` commands to update the yaml file just like flexy-template code

Reply:
Thank you @jianlinliu let me try 2.run python commands to update the yaml file just like flexy-template code

Reply:
I tried 2.run python commands to update the yaml file just like flexy-template code, but it reported `ModuleNotFoundError: No module named 'yaml'`, tried to install it, but reported `pip: command not found`, so I changed to use `sed` and it worked as expected. @jianlinliu it this ok?
```
sh-4.4$ /cli/oc get machine -n openshift-machine-api
NAME                                                 PHASE     TYPE          REGION      ZONE         AGE
ci-op-3hdqjjqn-d8ee5-7pxt5-master-0                  Running   m6i.xlarge    us-east-2   us-east-2a   66m
ci-op-3hdqjjqn-d8ee5-7pxt5-master-1                  Running   m6i.xlarge    us-east-2   us-east-2b   66m
ci-op-3hdqjjqn-d8ee5-7pxt5-master-2                  Running   m6i.xlarge    us-east-2   us-east-2c   66m
ci-op-3hdqjjqn-d8ee5-7pxt5-worker-us-east-2a-dtntt   Running   c5n.9xlarge   us-east-2   us-east-2a   60m
ci-op-3hdqjjqn-d8ee5-7pxt5-worker-us-east-2b-2txqh   Running   m5.xlarge     us-east-2   us-east-2b   60m
ci-op-3hdqjjqn-d8ee5-7pxt5-worker-us-east-2c-wts4h   Running   m5.xlarge     us-east-2   us-east-2c   60m
sh-4.4$ /cli/oc edit machine ci-op-3hdqjjqn-d8ee5-7pxt5-worker-us-east-2a-dtntt -n openshift-machine-api
...
      instanceType: c5n.9xlarge
      kind: AWSMachineProviderConfig
      metadata:
        creationTimestamp: null
      metadataServiceOptions: {}
      networkInterfaceType: EFA
      placement:
        availabilityZone: us-east-2a
        region: us-east-2
      placementGroupName: pgcluster
...a
```
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/41990/rehearse-41990-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-efa-pg-f14/1689565389450645504

Reply:
`sed` is also okay for me, but before that, can we try one more command - `pip3 install pyyaml --user`?

Reply:
Ok, let me try `pip3` first.